zerver/lib/validator.py

```diff
@@ -217,29 +217,6 @@ def f(var_name: str, val: object) -> List[ResultT]:
     return f
 
 
-def check_tuple(sub_validators: List[Validator[ResultT]]) -> Validator[Tuple[Any, ...]]:
-    def f(var_name: str, val: object) -> Tuple[Any, ...]:
-        if not isinstance(val, tuple):
-            raise ValidationError(_("{var_name} is not a tuple").format(var_name=var_name))
-
-        desired_len = len(sub_validators)
-        if desired_len != len(val):
-            raise ValidationError(
-                _("{var_name} should have exactly {desired_len} items").format(
-                    var_name=var_name,
-                    desired_len=desired_len,
-                )
-            )
-
-        for i, sub_validator in enumerate(sub_validators):
-            vname = f"{var_name}[{i}]"
-            sub_validator(vname, val[i])
-
-        return val
-
-    return f
```

Comment:
These validators can be used by code other than JSON decoding.  E.g. the last use was in 4f3d5f2d871597c7cd9a61f7eab3bdead3ceaa64, which is verifying event formats.  I think it's reasonable to remove given that we've removed tuples from the API in general.